edb/lib/std/26-bitwisefuncs.edgeql

```diff
@@ -0,0 +1,241 @@
+    USING SQL $$
+    SELECT l # r
+    $$;
+};
+
+
+CREATE FUNCTION
+std::bit_xor(l: std::int32, r: std::int32) -> std::int32
+{
+    CREATE ANNOTATION std::description :=
+        'Bitwise exclusive OR operator for 32-bit integers.';
+    SET volatility := 'Immutable';
+    USING SQL $$
+    SELECT l # r
+    $$;
+};
+
+
+CREATE FUNCTION
+std::bit_xor(l: std::int64, r: std::int64) -> std::int64
+{
+    CREATE ANNOTATION std::description :=
+        'Bitwise exclusive OR operator for 64-bit integers.';
+    SET volatility := 'Immutable';
+    USING SQL $$
+    SELECT l # r
+    $$;
+};
+
+
+CREATE FUNCTION
+std::bit_not(r: std::int16) -> std::int16
+{
+    CREATE ANNOTATION std::description :=
+        'Bitwise NOT operator for 16-bit integers.';
+    SET volatility := 'Immutable';
+    USING SQL $$
+    SELECT ~r
+    $$;
+};
+
+
+CREATE FUNCTION
+std::bit_not(r: std::int32) -> std::int32
+{
+    CREATE ANNOTATION std::description :=
+        'Bitwise NOT operator for 32-bit integers.';
+    SET volatility := 'Immutable';
+    USING SQL $$
+    SELECT ~r
+    $$;
+};
+
+
+CREATE FUNCTION
+std::bit_not(r: std::int64) -> std::int64
+{
+    CREATE ANNOTATION std::description :=
+        'Bitwise NOT operator for 64-bit integers.';
+    SET volatility := 'Immutable';
+    USING SQL $$
+    SELECT ~r
+    $$;
+};
+
+
+# In Postgres bitwise shift operators accept a 32-bit integer as the number of
+# bit positions that need to be shifted. However, in EdgeDB the default
+# integer literal is int64, so we should accept that and cast it down inside
+# the function body.
+CREATE FUNCTION
+std::bit_rshift(val: std::int16, n: std::int64) -> std::int16
+{
+    CREATE ANNOTATION std::description :=
+        'Bitwise shift right operator for 16-bit integers.';
+    SET volatility := 'Immutable';
+    USING SQL $$
+    SELECT val >> n::int4
+    $$;
+};
+
+
+CREATE FUNCTION
+std::bit_rshift(val: std::int32, n: std::int64) -> std::int32
+{
+    CREATE ANNOTATION std::description :=
+        'Bitwise shift right operator for 32-bit integers.';
+    SET volatility := 'Immutable';
+    USING SQL $$
+    SELECT val >> n::int4
+    $$;
+};
+
+
+CREATE FUNCTION
+std::bit_rshift(val: std::int64, n: std::int64) -> std::int64
+{
+    CREATE ANNOTATION std::description :=
+        'Bitwise shift right operator for 64-bit integers.';
+    SET volatility := 'Immutable';
+    USING SQL $$
+    SELECT val >> n::int4
+    $$;
+};
+
+
+CREATE FUNCTION
+std::bit_lshift(val: std::int16, n: std::int64) -> std::int16
+{
+    CREATE ANNOTATION std::description :=
+        'Bitwise shift left operator for 16-bit integers.';
+    SET volatility := 'Immutable';
+    USING SQL $$
+    SELECT val << n::int4
+    $$;
+};
+
+
+CREATE FUNCTION
+std::bit_lshift(val: std::int32, n: std::int64) -> std::int32
+{
+    CREATE ANNOTATION std::description :=
+        'Bitwise shift left operator for 32-bit integers.';
+    SET volatility := 'Immutable';
+    USING SQL $$
```

Comment:
As discussed on the call, this needs some extra handling for big vals of `n`. Also negative `n` shouldn't be allowed neither for rshift nor for lshit